Tests/SwiftDocCTests/Benchmark/TopicGraphHashTests.swift

```diff
@@ -114,8 +112,8 @@ class TopicGraphHashTests: XCTestCase {
         XCTAssertEqual(taskGroupLinks, [
             "doc://org.swift.docc.example/documentation/Test-Bundle/article",
             "doc://org.swift.docc.example/documentation/Test-Bundle/article2",
-            "doc://com.external.testbundle/article",
-            "doc://com.external.testbundle/article2",
```

Comment:
This test is an example of the bug that was possible with the previous code. These "expected" values are actually the _unresolved_ external links. 

The bug in this test is that the configuration (the external documentation source) is added _after_ registering the data provider with the workspace and context.

```swift
try workspace.registerProvider(dataProvider)
let context = try DocumentationContext(dataProvider: workspace)

// At this point the context has already "built" all documentation
//
// Any changes after this point has no effect. 
// Specifically, any external links have already failed to resolve and the context won't try again.

context.externalDocumentationSources = ["com.external.testbundle" : ExternalReferenceResolverTests.TestExternalReferenceResolver()]
```